src/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -198,7 +198,16 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
   public addItem(doScroll: boolean): void {
     // Manually resetting errors is required to prevent `ExpressionChangedAfterItHasBeenCheckedError`
     this.formArray.setErrors(null);
-    const item = { value: null }
+    let item = { value: null }
+
+    // Clonning last item and incrementing Id
+    if (this.caseField.field_type && this.caseField.field_type.collection_field_type
```

Comment:
So I don't think you need to do all this, you could just call the method :
`this.caseField.isDynamic()`